File: frontend/lib/models/restaurant/table.dart
// lib/models/restaurant/table.dart

class RestaurantTable {
  final String? id;
  final String name;
  final int capacity;

  RestaurantTable({
    this.id,
    required this.name,
    required this.capacity,
  });

  factory RestaurantTable.fromJson(Map<String, dynamic> json) {
    return RestaurantTable(
      id: json['id'],
      name: json['name'],
      capacity: json['capacity'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'capacity': capacity,
    };
  }
}

File: backend/models/restaurant/table.py

# backend/models/restaurant/table.py

from sqlalchemy import Column, String, Integer
from models.base import Base
import uuid

class RestaurantTable(Base):
    __tablename__ = "restaurant_tables"

    id = Column(String, primary_key=True, index=True, default=lambda: str(uuid.uuid4()))
    name = Column(String, nullable=False)
    capacity = Column(Integer, nullable=False)
